nima/webclient/webclient/src/test/java/io/helidon/nima/webclient/HttpClientTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.nima.webclient;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.function.Function;
+
+import io.helidon.common.http.ClientRequestHeaders;
+import io.helidon.common.http.Http;
+import io.helidon.common.http.WritableHeaders;
+import io.helidon.nima.common.tls.Tls;
+import io.helidon.nima.webclient.http1.Http1ClientResponse;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+class HttpClientTest {
```

Comment:
My personal opinion here is that this is 150+ lines to test something extremely simple